### PR TITLE
dev/core#818 Partial conversion of Group form to be an EntityForm

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code cleanup - Partial form conversion to entity form

Before
----------------------------------------
Less shared code, less code conventions followed

After
----------------------------------------
More shared code , more code conventions followed. Metadata improved.

This does slightly change the field name - 'Name' becomes 'Group Title' - it's possible to override name changes but I only do that if the one from the metadata is clearly worse not much of a muchness. Likewise on delete the 'Delete Group' button becomes the 'Delete' button - which is consistent with elsewhere

Screenshot 2019-03-25 22 35 11

Technical Details
----------------------------------------
This is mostly for demo purposes and it should be noted that these changes are generally done in a'step in the right direction' kinda way, and in accordance with this I have made a very minimal change.

 In this case the change gets the form to be using the entity form and converts two fields. The change didn't initially work for the title field because the schema didn't define the
html type for the field - fixing these things is a feature not a bug of switching to an entity form

We can remove the apply filter line  & the addButtons fn as the parent does these (actually it adds the custom
data too but I've left that out of scope).

If we converted a few more fields we would see that
1) the saved_search_id doesn't have a pseudoconstant defined in the schema, as it should be - figuring out how to clean that up would mean that metadata is available to the api as well as to a replacement form layer
2) the various field description & help info is not in the schema
3) when we convert the tpl file we add custom data support in, although this form already has it

When we finally switch to a non-quickform layer the new forms will be built of the schema so by
transferring information into the schema and out of thousands of differently coded forms
we take steps in the direction of facilitating that.

That perhaps seems a little unachievable but the api was originally as incoherent as the form
layer and it was only by incremental cleanup it became more consistent

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Group/Form/Edit.php [hunk 1]
```diff
@@ -78,6 +80,35 @@ class CRM_Group_Form_Edit extends CRM_Core_Form {
...
+    ];
+  }
+
+  /**
+   * Set the delete message.
+   *
+   * We do this from the constructor in order to do a translation.
+   */
+  public function setDeleteMessage() {
+    $this->deleteMessage = '';
```
Comment (eileenmcnaughton) [inline CRM/Group/Form/Edit.php]: This is still coming from the tpl at this stage

**Comment by eileenmcnaughton** [comment]
@mattwire can you check this at some point ?

**Review by pfigel — APPROVED** [review]
(*CiviCRM Review Template WORD-1.2*)



* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __PASS__
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __PASS__: Form field labels changed as stated in description, but are still clear.
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __PASS__
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __PASS__: Created new groups (and smart groups) with and without this patch. Updated existing groups. No errors encountered in the UI, and found no differences in the database when comparing old vs. new in `civicrm_group`.
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __PASS__: `CRM_Contact_Form_Task_SaveSearch` has some calls to methods in `CRM_Group_Form_Edit` , but those are still compatible (and were covered by `r-run`). Impact to extensions having e.g. form hooks on that form unlikely as field *names* didn't change.
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __PASS__
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __PASS__
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __PASS__

**Comment by eileenmcnaughton** [comment]
Thanks @pfigel !!

**Pull request merged**